doc/dev/kernel-client-troubleshooting.rst

```diff
@@ -9,8 +9,9 @@ will be output in dmesg. Collect it and any appropriate kernel state. If
 the problem is with the MDS, there will be hung requests that the client
 is waiting on. Look in ``/sys/kernel/debug/ceph/*/`` and cat the ``mdsc`` file to get a listing of requests in progress. If one of them remains there, the
 MDS has probably "forgotten" it.
-We can get hints about what's going on by dumping the MDS cache:
-ceph mds tell 0 dumpcache /tmp/dump.txt
+We can get hints about what's going on by dumping the MDS cache ::
+
+  ceph daemon mds.<id> dump cache > /tmp/dump.txt
```

Comment:
It's not recommended to dump the cache to stdout as it is usually too large. Instead just do:

`... dump cache /tmp/dump.txt`

and note that the file is on the machine executing the MDS. Also, if run via systemd, it's located in the namespace of the MDS. It needs to be fetched via the `nsenter(1)` command. Probably, you should create a dedicated section in the `doc/cephfs/troubleshooting.rst` and link to that.

Reply:
I will do that.

Reply:
@batrick After reading `doc/cephfs/troubleshooting.rst`, I think the [entire file](https://github.com/ceph/ceph/blob/master/doc/dev/kernel-client-troubleshooting.rst) could be dropped and then modify the following sections in `doc/cephfs/troubleshooting.rst` appropriately, so that everything would be at one place. The `kernel-client-troubleshooting.rst` is not linked anywhere, so there shouldn't be other problems.

- http://docs.ceph.com/docs/master/cephfs/troubleshooting/#kernel-mount-debugging
- http://docs.ceph.com/docs/master/cephfs/troubleshooting/#the-mds

What do you think?

Reply:
I'm okay with some reorganizing of the docs. However, I think the kernel debugging docs would be better part of a larger doc, like [1]. However, [1]'s title in the index is not clear that it's for the kernel. That should be fixed too.

[1] http://docs.ceph.com/docs/master/cephfs/kernel